base/math.jl

```diff
@@ -146,15 +146,17 @@ sqrt(x::Float32) = box(Float32,sqrt_llvm(unbox(Float32,x)))
 sqrt(x::Real) = sqrt(float(x))
 @vectorize_1arg Number sqrt
 
-hypot(x::Real, y::Real) = hypot(promote(float(x), float(y))...)
-function hypot{T<:AbstractFloat}(x::T, y::T)
+@inline hypot(x::Real, y::Real) = hypot(promote(float(x), float(y))...)
+@inline function hypot{T<:AbstractFloat}(x::T, y::T)
     x = abs(x)
     y = abs(y)
     if x < y
         x, y = y, x
     end
-    if x == 0
+    if x == zero(x)
         r = y/one(x)
+    elseif x < sqrt(realmax(x))
```

Comment:
this might need to be `sqrt(realmax(x)/2)`, in case `y` is the same magnitude as `x` (e.g. `x = y =prevfloat(realmax(Float64))`)